internal/metrics/prometheus.go

```diff
@@ -0,0 +1,170 @@
+package metrics
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	ctrlmetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+// Recorder is the interface for recording metrics for provided operation.
+type Recorder interface {
+	RecordKonnectEntityOperationSuccess(serverURL string, operationType KonnectEntityOperation, entityType string, duration time.Duration)
+	RecordKonnectEntityOperationFailure(serverURL string, operationType KonnectEntityOperation, entityType string, duration time.Duration, statusCode int)
+}
+
+type KonnectEntityOperation string
+
+const (
+	KonnectServerURLKey                                  = "server_url"
+	KonnectEntityOperationTypeKey                        = "operation_type"
+	KonnectEntityOperationCreate  KonnectEntityOperation = "create"
+	KonnectEntityOperationUpdate  KonnectEntityOperation = "update"
+	KonnectEntityOperationDelete  KonnectEntityOperation = "delete"
+
+	KonnectEntityTypeKey = "entity_type"
+
+	SuccessKey   = "success"
+	SuccessTrue  = "true"
+	SuccessFalse = "false"
+
+	StatusCodeKey = "status_code"
+)
```

Comment:
Ideally we'd add godocs for these `const`s

Reply:
Added comments.